### PR TITLE
Jordan/1366 decimals

### LINKED ISSUE #1366: Show all decimals for amounts
Description:

~Show only 2 decimals~ (JB EDIT - SHOW ALL DECIMALS)

screen shot 2018-09-26 at 6 42 25 pm

### PR DESCRIPTION
**Pull request by jbibla**
Closes #1366 

Description:

- update num script to use 10 decimals
- display 10 decimal places where we show bonding denom balance

❤️ Thank you!

### CODE REVIEW

**Comment by faboweb** [comment]
What about the idea with the small decimals?

File: app/src/renderer/components/staking/LiValidator.vue [hunk 1]
```diff
@@ -9,7 +9,7 @@ li.li-validator(:class='styles')
         .top {{ validator.description.moniker }}
         .bottom {{ shortAddress(validator.id)}}
     .li-validator__value.your-votes
-      span {{ yourVotes.isLessThan(0.01) && yourVotes.isGreaterThan(0) ? '< ' + num.pretty(0.01) : num.pretty(yourVotes) }}
+      span {{ yourVotes.isLessThan(0.01) && yourVotes.isGreaterThan(0) ? '< ' + num.full(0.01) : num.full(yourVotes) }}
```
[the remark this answers is not in the document]
Comment (jbibla) [reply app/src/renderer/components/staking/LiValidator.vue]: it does

File: app/src/renderer/components/staking/PageValidator.vue [hunk 2]
```diff
@@ -32,7 +32,7 @@ tm-page
         .row.validator-profile__header__data
           dl.colored_dl
             dt Bonded {{bondingDenom}}
-            dd {{myBond.isLessThan(0.01) && myBond.isGreaterThan(0) ? '< ' + 0.01 : pretty(myBond)}}
+            dd {{myBond.isLessThan(0.01) && myBond.isGreaterThan(0) ? '< ' + 0.01 : num.full(myBond)}}
```
[the remark this answers is not in the document]
Comment (jbibla) [reply app/src/renderer/components/staking/PageValidator.vue]: it does

**Comment by jbibla** [comment]
> What about the idea with the small decimals?

i experimented with this but could not land on something i felt was better than keeping them the same size. let's merge this and if we want in the future we can always improve it.

**Comment by jbibla** [comment]
@NodeGuy 100% coverage for TabMyDelegations 🕺

File: test/unit/specs/components/staking/TabMyDelegations.spec.js [hunk 3]
```diff
@@ -151,7 +155,77 @@ test(`yourValidators`, () => {
...
+    })
+  ).toEqual(`cosmos1asdfasdf`)
+})
+
+test(`properties`, () => {
+  expect(
+    TabMyDelegations.computed.properties({
+      bondingDenom: `tofu-steaks`
+    })
+  ).toEqual([
```
Comment (faboweb) [inline test/unit/specs/components/staking/TabMyDelegations.spec.js]: Good example on where to use a snapshot.

Comment (jbibla) [reply test/unit/specs/components/staking/TabMyDelegations.spec.js]: i don't know how in @NodeGuy's brave new world.

also, the amount of unused code was _not good_.

Comment (faboweb) [reply test/unit/specs/components/staking/TabMyDelegations.spec.js]: ?

Comment (jbibla) [reply test/unit/specs/components/staking/TabMyDelegations.spec.js]: where does the `wrapper` come from without `vuex-setup` and `mount`?

```
expect(wrapper.html()).toMatchSnapshot()
```

Comment (faboweb) [reply test/unit/specs/components/staking/TabMyDelegations.spec.js]: there is still DOM generated in `shallow`. Components are just not resolved so they stay as `<componen-x prop1="2" />`.
In this case using a snapshot means: 
```
expect(
    TabMyDelegations.computed.properties({
      bondingDenom: `tofu-steaks`
    })
  ).toMatchSnapshot()
```

Comment (jbibla) [reply test/unit/specs/components/staking/TabMyDelegations.spec.js]: ah cool. thank you!

Comment (faboweb) [reply test/unit/specs/components/staking/TabMyDelegations.spec.js]: this way we don't pollute the test file with the huge json but still see if anything changes to the unexpected.

**Pull request merged**